### ISSUE TITLE
[SyliusUiBundle] Decouple from the Sylius stack

### ISSUE THREAD

**Issue by bpolaszek** [Sylius/Sylius#7536]
Hello everyone,

Tl;DR:
-----
**SyliusUiBundle** misses a few refactoring steps to be used as a standalone in conjunction with **SyliusResourceBundle** and **SyliusGridBundle**. These 3 bundles could be used in any Symfony project to create a fully-functionnal back office with a modern and responsive user interface in a few steps.

The main point
-----

I'm currently creating a big admin project from scratch and I want to use **SyliusResourceBundle** and **SyliusGridBundle**.

These 2 bundles provide a quick way of exposing an API with the entities, and to setup some grid views with filters, pagination, action buttons, etc.

The problem is **SyliusGridBundle** only provides the configuration part, and not any twig file.
**SyliusUiBundle** has been created to use **SyliusResourceBundle** and **SyliusGridBundle** to render an admin UI based on the _Semantic UI_ front-end framework.

I know **SyliusUiBundle**  is part of the Sylius stack and is not intended to be used as standalone.

But actually, it provides many basic resources needed in any admin project:
* Advanced theming of forms, modals, grids, breadcrumbs, etc
* Translations of many common used labels in back office interfaces (create, edit, address, etc)
* JS interactions (confirmation messages, autocomplete, ...)

So, I have tried to use it as a standalone into my project and here are the few obstacles I had to face:
* Unknown "sylius_currency_symbol" filter in SyliusUiBundle:Form:theme.html.twig - this filter depends on SyliusCurrencyBundle
* Same for "sylius_locale_name" in the same file - this filter depends on SyliusLocaleBundle
* Twig files are referenced with the 'SyliusUiBundle::' notation instead of '@SyliusUi/', preventing them to be overriden
* Some JS like _sylius-product-attributes.js_  target sylius-specific DOM ids.

Most of the Sylius front implementations have been moved to the **SyliusAdminBundle**, but I think the work is not over.

What to do
------
I would like to help in achieving the standalone goal.

* A big find/replace to change the twig file notation should be easy with no BC break (tell me if I'm wrong) - that would allow any Sylius Twig file to be overridden in _app/Resources_ or by bundle inheritance.
* That said, should **SyliusAdminBundle** have **SyliusUiBundle** as parent? I think it would be logical, but I don't know in advance the BC breaks and the issues this could cause.
* I'm really not familiar with the JS part (gulp, etc) so I won't try to fix the _sylius-product-attributes.js_
* SyliusUIBundle does not work out of the box from the assets point of view: I had to import semantic-ui by myself, and some CSS rules are broken. A more generic approach may be adopted, and the documentation for having all the CSS and JS to work may be necessary.

What's your opinion?

**Comment by michalmarcinkowski** [comment]
Related issues https://github.com/Sylius/Sylius/issues/5236, https://github.com/Sylius/Sylius/issues/6473.
